source/code/plugins/oms_diag_lib.rb

```diff
@@ -24,8 +24,78 @@ class Diag
     # Record constant values
     RECORD_MGID_VALUE = '{00000000-0000-0000-0000-000000000002}'
 
+    # Diagnostic logging support minimum version
+    DIAG_MIN_VERSION = "1.3.4-127"
+
     class << self
 
+      @@IsDiagSupported = nil
+      @@InstallInfoPath = "/etc/opt/microsoft/omsagent/sysconf/installinfo.txt"
+
+      # Method to check if diagnostic logging is supported
+      #
+      # Description:
+      # This would tell if current omsagent version supports
+      # diagnostic logging
+      #
+      # NOTE: If this returns false then logs will be rejected silently
+      def IsDiagEnabled()
+        return @@IsDiagSupported unless @@IsDiagSupported.nil?
+
+        begin
+          # Read installinfo.txt
+          versionline = IO.readlines(@@InstallInfoPath)[0]
+
+          # Extract version number
+          versionnum = versionline.split()[0]
+
+          # Extract major and minor parts of version number
+          major, minor = GetVersionAsArray(versionnum)
+
+          # Check validity of major and minor parts
+          if major.nil? or
+             !major.is_a?(Array)
+             major.size() != 3 or
+             minor.nil? or
+             !minor.is_a?(String)
+            @@IsDiagSupported = false
+            return @@IsDiagSupported
+          end
+
+          # Compare version number
+          min_major, min_minor = GetVersionAsArray(DIAG_MIN_VERSION)
+
+          # Compare major parts
+          for x in 0..2
```

Comment:
Checked the logic. It is the same. Just that I would have to make it major, minor patch and build. Will update right away.